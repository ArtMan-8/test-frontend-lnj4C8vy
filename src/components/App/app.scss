.app {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100vw;
  height: 100vh;

  background: linear-gradient(255.35deg, var(--color-red1) 0.83%, rgba(255, 79, 79, 0) 108.93%), var(--color-red2);
}

.app__btn-ded {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 12px 24px;

  font-weight: 500;
  font-size: 12px;
  color: var(--color-white);

  background: transparent;
  border: 1px solid var(--color-white);
  border-radius: 6px;
  cursor: pointer;
  filter: drop-shadow(0px 0px 44px var(--color-light));

  &:hover {
    color: var(--color-black);

    background: var(--color-white);
  }

  &:active {
    color: var(--color-black);

    background: var(--color-white);
  }

  &:disabled {
    color: var(--color-white);

    background: var(--color-grey);
    border-color: var(--color-grey);
  }
}
