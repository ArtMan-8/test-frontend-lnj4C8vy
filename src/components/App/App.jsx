import React, { useState } from 'react';
import Popup from '../Popup';
import './app.scss';

const App = () => {
  const [isPopup, setIsPopup] = useState(false);

  return (
    <section className="app">
      <button
        type="button"
        className="app__btn-ded"
        onClick={() => setIsPopup(true)}
        disabled={isPopup}
      >Налоговый вычет</button>

      {isPopup && <Popup handleClosePopup={() => setIsPopup(false)} />}
    </section>
  );
};

export default App;
