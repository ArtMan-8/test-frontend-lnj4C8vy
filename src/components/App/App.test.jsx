import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App.jsx';

afterEach(cleanup);

test('App snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
