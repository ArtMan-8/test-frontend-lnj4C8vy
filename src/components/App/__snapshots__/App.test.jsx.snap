// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<DocumentFragment>
  <section
    class="app"
  >
    <button
      class="app__btn-ded"
      type="button"
    >
      Налоговый вычет
    </button>
  </section>
</DocumentFragment>
`;
